### PR TITLE
Initial merge of some PTS v3 changes

### PR DESCRIPTION
**Pull request by Mazo**
Beginnings of the attempt to bring this repo and @SouNourS PlayTest fork into sync somewhat.

Many things aren't included in this initial merge, notably anything database/account related (amongst many other things / tweaks that should stay on PTSv3 only)

### CODE REVIEW

File: common/src/main/scala/net/psforever/types/BailType.scala [hunk 1]
```diff
@@ -8,7 +8,13 @@ object BailType extends Enumeration {
   type Type = Value
 
   val Normal = Value(0)
+  val Unk1   = Value(1) // to have Xtoolspar working
```
Comment (pschord) [inline common/src/main/scala/net/psforever/types/BailType.scala]: Do we have a link to this tool?

File: common/src/main/scala/services/chat/ChatAction.scala [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2017 PSForever
+package services.chat
+
+import net.psforever.objects.zones.Zone
+import net.psforever.packet.game.{ChatMsg, PlanetSideGUID}
+import net.psforever.types.{PlanetSideEmpire, Vector3}
+
+object ChatAction {
+  trait Action
```
Comment (pschord) [inline common/src/main/scala/services/chat/ChatAction.scala]: can we make this a sealed trait?

File: common/src/main/scala/services/chat/ChatResponse.scala [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2017 PSForever
+package services.chat
+
+import net.psforever.packet.game.PlanetSideGUID
+import net.psforever.types.ChatMessageType
+
+object ChatResponse {
+  trait Response
```
Comment (pschord) [inline common/src/main/scala/services/chat/ChatResponse.scala]: same here

File: pslogin/src/main/scala/WorldSessionActor.scala [hunk 4]
```diff
@@ -88,13 +90,24 @@ class WorldSessionActor extends Actor with MDCContextAware {
...
+  var whenUsedLastAIMAX : Long = 0
+  var whenUsedLastAVMAX : Long = 0
+  var whenUsedLastMAX : Array[Long] = Array.fill[Long](4)(0L)
+  var whenUsedLastMAXName : Array[String] = Array.fill[String](4)("")
+  var whenUsedLastItem : Array[Long] = Array.fill[Long](1020)(0L)
+  var whenUsedLastItemName : Array[String] = Array.fill[String](1020)("")
   var whenUsedLastKit : Long = 0
+  var whenUsedLastSMKit : Long = 0
+  var whenUsedLastSAKit : Long = 0
+  var whenUsedLastSSKit : Long = 0
```
Comment (pschord) [inline pslogin/src/main/scala/WorldSessionActor.scala]: We definitely need to improve this on a followup patch. Lots of duplication

File: pslogin/src/main/scala/WorldSessionActor.scala [hunk 5]
```diff
@@ -4117,6 +4389,76 @@ class WorldSessionActor extends Actor with MDCContextAware {
                         }
                       }
                     }
+                    else if(kit.Definition == GlobalDefinitions.super_medkit) {
+                      if(player.Health == player.MaxHealth) {
+                        sendResponse(ChatMsg(ChatMessageType.UNK_225, false, "", "@HealComplete", None))
+                      }
+                      else if(System.currentTimeMillis - whenUsedLastSMKit < 1200000) {
```
Comment (pschord) [inline pslogin/src/main/scala/WorldSessionActor.scala]: Yikes. All of the magic comparisons to millis will need to go on future patches. We need a nice timer class

Comment (pschord) [reply pslogin/src/main/scala/WorldSessionActor.scala]: By "go" meaning we should fix them up later, not as in on this PR

File: common/src/main/scala/net/psforever/types/BailType.scala (hunk 1 above)
Comment (Fate-JH) [reply common/src/main/scala/net/psforever/types/BailType.scala]: Which version do you want?  Sounours wrote the original (`XTools`) and I think between Sounours, Mazo, and myself, we each made unshared modifications to it for our own purposes.

It's a packet capture decoder.

File: pslogin/src/main/scala/WorldSessionActor.scala (hunk 5 above)
Comment (Mazo) [reply pslogin/src/main/scala/WorldSessionActor.scala]: Agreed

File: pslogin/src/main/scala/WorldSessionActor.scala (hunk 4 above)
Comment (Mazo) [reply pslogin/src/main/scala/WorldSessionActor.scala]: Definitely. I didn't want to make any changes to soun's code until we've got the repos more closely aligned

File: pslogin/src/main/scala/WorldSessionActor.scala [hunk 6]
```diff
@@ -88,13 +90,24 @@ class WorldSessionActor extends Actor with MDCContextAware {
   var flying : Boolean = false
   var speed : Float = 1.0f
   var spectator : Boolean = false
-  var admin : Boolean = false
+  var admin : Boolean = true
```
[the remark this answers is not in the document]
Comment (SouNourS) [reply pslogin/src/main/scala/WorldSessionActor.scala]: Maybe he wanted try GM things, on mine it's false, but linked to DB.
https://github.com/SouNourS/PSF-LoginServer/blob/d69f1b94e3d1c24893e414ca48ac39e4f8553c43/pslogin/src/main/scala/WorldSessionActor.scala#L91

File: pslogin/src/main/scala/WorldSessionActor.scala (hunk 4 above)
Comment (SouNourS) [reply pslogin/src/main/scala/WorldSessionActor.scala]: @Mazo you can play, I'm a noob ;)

File: pslogin/src/main/scala/WorldSessionActor.scala (hunk 6 above)
[the remark this answers is not in the document]
Comment (Mazo) [reply pslogin/src/main/scala/WorldSessionActor.scala]: Yikes. Must have been testing something and forgot to flip it back. Will amend.

File: common/src/main/scala/net/psforever/objects/Avatar.scala [hunk 7]
```diff
@@ -12,6 +12,8 @@ import scala.annotation.tailrec
 import scala.collection.mutable
 
 class Avatar(val name : String, val faction : PlanetSideEmpire.Value, val sex : CharacterGender.Value, val head : Int, val voice : CharacterVoice.Value) {
+  /** Character ID from the DB */
+  private var charId : Long = 0
```
[the remark this answers is not in the document]
Comment (Mazo) [reply common/src/main/scala/net/psforever/objects/Avatar.scala]: I did notice he added character id in his branch too. @SouNourS I assume you've made changes on your fork to merge the two already?

[the remark this answers is not in the document]
Comment (SouNourS) [reply common/src/main/scala/net/psforever/objects/Avatar.scala]: Yes sir !
https://github.com/SouNourS/PSF-LoginServer/blob/PlayTestServerV3/common/src/main/scala/net/psforever/objects/Avatar.scala

File: pslogin/src/main/scala/WorldSessionActor.scala (hunk 6 above)
[the remark this answers is not in the document]
Comment (Mazo) [reply pslogin/src/main/scala/WorldSessionActor.scala]: Amended.

File: common/src/main/scala/services/chat/ChatAction.scala (hunk 2 above)
Comment (Mazo) [reply common/src/main/scala/services/chat/ChatAction.scala]: Amended.

File: common/src/main/scala/services/chat/ChatResponse.scala (hunk 3 above)
Comment (Mazo) [reply common/src/main/scala/services/chat/ChatResponse.scala]: Amended.

File: common/src/main/scala/net/psforever/types/BailType.scala (hunk 1 above)
Comment (pschord) [reply common/src/main/scala/net/psforever/types/BailType.scala]: I was just wondering if it was on Github

**Pull request merged**